src/app/Fake.Tools.Squirrel/Squirrel.fs

```diff
@@ -0,0 +1,143 @@
+
+        /// The path to Squirrel: `squirrel.exe`
+        ToolPath : string
+
+        /// Maximum time to allow Squirrel to run before being killed.
+        TimeOut : TimeSpan
+
+        /// Sign the installer with signtool.exe
+        SignExecutable : bool option
+
+        /// The code signing certificate to be used for signing
+        SigningKeyFile : string option
+
+        /// The secret key for the code signing certificate
+        SigningSecret : string option }
+
+let internal defaultParams = lazy(
+    let toolname = "Squirrel.exe"
+    {
+        ReleaseDir = ""
+        WorkingDir = None
+        BootstrapperExe = None
+        LoadingGif = None
+        SetupIcon = None
+        NoMsi = false
+        ToolPath = Tools.findToolInSubPath toolname ( Directory.GetCurrentDirectory() </> "tools" </> "Squirrel")
+        TimeOut = TimeSpan.FromMinutes 10.
+        SignExecutable = None
+        SigningKeyFile = None
+        SigningSecret = None })
+
+let private createSigningArgs (parameters : ReleasifyParams) =
+    new StringBuilder()
+    |> StringBuilder.appendWithoutQuotes "--signWithParams=\""
+    |> StringBuilder.appendWithoutQuotes "/a"
+    |> StringBuilder.appendIfSome parameters.SigningKeyFile (sprintf "/f %s")
+    |> StringBuilder.appendIfSome parameters.SigningSecret  (sprintf "/p %s")
+    |> StringBuilder.appendWithoutQuotes "\""
+    |> StringBuilder.toText
+
+let internal buildSquirrelArgs parameters nugetPackage =
+    new StringBuilder()
+    |> StringBuilder.appendIfNotNullOrEmpty nugetPackage "--releasify="
+    |> StringBuilder.appendIfNotNullOrEmpty parameters.ReleaseDir "--releaseDir="
+    |> StringBuilder.appendIfSome parameters.LoadingGif (sprintf "\"--loadingGif=%s\"")
+    |> StringBuilder.appendIfSome parameters.SetupIcon (sprintf "\"--setupIcon=%s\"")
+    |> StringBuilder.appendIfTrue parameters.NoMsi "--no-msi"
+    |> StringBuilder.appendIfSome parameters.BootstrapperExe (sprintf "\"--bootstrapperExe=%s\"")
+    |> StringBuilder.appendIfSome parameters.SignExecutable (fun _ -> createSigningArgs parameters)
+    |> StringBuilder.toText
+
+module internal ResultHandling =
+    let (|OK|Failure|) = function
+        | 0 -> OK
+        | x -> Failure x
+
+    let buildErrorMessage = function
+        | OK -> None
+        | Failure errorCode ->
+            Some (sprintf "Squirrel reported an error (Error Code %d)" errorCode)
+
+    let failBuildIfSquirrelReportedError =
+        buildErrorMessage
+        >> Option.iter failwith
+
+/// Creates a Squirrel installer for given NuGet package
+/// Will fail if Squirrel terminates with non-zero exit code.
+///
+/// ## Parameters
+///
+///  - `setParams` - Function used to manipulate the default `SquirrelParams` value.
+///  - `nugetPackage` - The package to create an installer for
+///
+/// ## Sample usage
+///
+///     Target.create "CreatePackage" (fun _ ->
+///         Squirrel.pack (fun p -> { p with WorkingDir = Some "./tmp" }) "./my.nupkg"
```

Comment:
`Squirrel.pack` doesn't exist?

Reply:
That was the name I choose before realizing that their documentation use releasify everywhere